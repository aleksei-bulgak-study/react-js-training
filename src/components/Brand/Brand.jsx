import React from 'react';
import './Brand.css';

const Brand = () => (
  <p className="brand">
    <span className="brand__name">netflix</span>
    roulette
  </p>
);

export default Brand;
