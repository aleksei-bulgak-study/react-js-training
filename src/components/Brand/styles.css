.brand {
  height: 100%;
  display: flex;
  align-items: center;
  color: var(--red);
  font-weight: 200;
  font-size: 1.5rem;
}

.brand__name {
  font-weight: 800;
}
