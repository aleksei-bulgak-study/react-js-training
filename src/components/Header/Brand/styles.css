.brand {
  height: 100%;
  display: flex;
  align-items: center;
}

.brand-title {
  color: var(--red);
  position: relative;
  font-weight: 200;
  font-size: 1.5rem;
}

.brand-title__name {
  font-weight: 800;
}
