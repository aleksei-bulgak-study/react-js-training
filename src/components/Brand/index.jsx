import React from 'react';
import './styles.css';

const Brand = () => (
  <p className="brand">
    <span className="brand__name">netflix</span>
    roulette
  </p>
);

export default Brand;
