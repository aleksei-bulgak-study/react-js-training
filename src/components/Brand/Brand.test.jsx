import { render } from 'enzyme';
import React from 'react';
import Brand from './Brand';

describe('when Brand', () => {
  test('then snapshot created', () => {
    const brandComponent = render(<Brand />);

    expect(brandComponent.find('.brand__name').text()).toEqual('netflix');
    expect(brandComponent).toMatchSnapshot();
  });
});
