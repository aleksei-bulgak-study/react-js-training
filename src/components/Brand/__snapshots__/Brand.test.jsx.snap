// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when Brand then snapshot created 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "brand",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "brand__name",
        },
        "children": Array [
          Object {
            "data": "netflix",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "data": "roulette",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "text",
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "data": "roulette",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "brand__name",
          },
          "children": Array [
            Object {
              "data": "netflix",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "span",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "text",
      },
    ],
    "name": "p",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
